src/main/java/me/ryanhamshire/GriefPrevention/DataStore.java

```diff
@@ -1039,20 +1039,46 @@ synchronized public void extendClaim(Claim claim, int newDepth)
 
         if (claim.parent != null) claim = claim.parent;
 
+        //keep track of old values in case ClaimModifiedEvent is cancelled.
+        Claim oldClaim = claim.clone();
+
+        //values in this map are double arrays with length 2,
+        //first value the lesserBoundaryY and second value the greaterBoundaryY.
+        Map<Claim, double[]> oldYMap = new HashMap<>();
+        oldYMap.put(claim, new double[]{ claim.lesserBoundaryCorner.getY(), claim.greaterBoundaryCorner.getY()});
+
         //adjust to new depth
         claim.lesserBoundaryCorner.setY(newDepth);
         claim.greaterBoundaryCorner.setY(newDepth);
         for (Claim subdivision : claim.children)
         {
+            oldYMap.put(subdivision, new double[]{ subdivision.lesserBoundaryCorner.getY(), subdivision.greaterBoundaryCorner.getY() });
             subdivision.lesserBoundaryCorner.setY(newDepth);
             subdivision.greaterBoundaryCorner.setY(newDepth);
-            this.saveClaim(subdivision);
         }
 
-        //save changes
-        this.saveClaim(claim);
-        ClaimModifiedEvent event = new ClaimModifiedEvent(claim, null);
+        //call the event
+        ClaimModifiedEvent event = new ClaimModifiedEvent(oldClaim, claim, null);
```

Comment:
I haven't visited this logic for years, is there any reason why we have to also call and check the event here too?

Reply:
I guess only for completeness. The`extendClaim()` method is called by GP, it makes sense to call the event, because a claim did change, just not by the user. Maybe we could split the event up into two seperate events (ClaimResizeEvent and ClaimExtendEvent)?

Reply:
Wait, in what case would a claim change? Perhaps the auto extend task, but that's only the lower y value, and as far as GP's concerned in the 2D space, it hasn't changed.

For now I'd not fire the event there, and if you do need to capture that case, a different PR would be preferred. Not sure if we should do something like expose the task, or just make another event that makes it clear this is the "automatic downward extending" of a claim.

Reply:
But wouldn't that break addons which use the ClaimModifiedEvent to detect these kind of GP changes? I did not add the event here, it was there already.

Reply:
This event is very new and I doubt any addons are using this, if any - and if they are, they're likely looking for the resize done by a player.

Reply:
Sure. Will create the other PR when this one is merged - need the Claim#clone() method.